components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/dbhandler/DBHandler.java

```diff
@@ -0,0 +1,172 @@
+     */
+    Connection getConnection(HikariDataSource dataSource) {
+        try {
+            if (dataSource != null) {
+                Connection conn = dataSource.getConnection();
+                conn.setAutoCommit(true);
+                return conn;
+            } else {
+                throw new RDBMSTableException("Datasource Could not be found. " +
+                        "Hence cannot initialize the status dashboard.");
+            }
+        } catch (SQLException e) {
+            throw new RDBMSTableException("Error initializing connection: " + e.getMessage(), e);
+        }
+    }
+
+
+    /**
+     * Insert worker data worker db.
+     *
+     * @return isSuccess
+     */
+    boolean insert(PreparedStatement stmt) {
+        try {
+            stmt.execute();
+            return true;
+        } catch (SQLException e) {
+            throw new RDBMSTableException("Attempted execution of query [" + stmt.toString() + "] produced an " +
+                    "exceptions" +
+                    " in " + stmt.toString(), e);
+        } finally {
+            try {
+                stmt.close();
+            } catch (SQLException e) {
+                //ignore
+            }
+        }
+    }
+
+    /**
+     * Delete workers data to worker db.
+     *
+     * @return isSuccess.
+     */
+    public boolean delete(PreparedStatement stmt) {
+        try {
+            stmt.execute();
+            stmt.close();
+            return true;
+        } catch (SQLException e) {
+            throw new RDBMSTableException(e.getMessage() + " in " + stmt.toString(), e);
+        }
+    }
+
+    /**
+     * Select worker from the worker DB.
+     *
+     * @return list of object.
+     */
+
+    ResultSet select(PreparedStatement stmt) {
+        ResultSet rs;
+        try {
+            rs = stmt.executeQuery();
+        } catch (SQLException e) {
+            throw new RDBMSTableException("Error retrieving records from table '" + stmt.toString() + "': "
+                    + e.getMessage(), e);
+        }
+        return rs;
+    }
+
+    /**
+     * Update workers in the worker db.
+     *
+     * @return new object that updated;
+     */
+    Object update(PreparedStatement stmt) {
+        try {
+            stmt.execute();
+            stmt.close();
+            return stmt;
+        } catch (SQLException e) {
+            throw new RDBMSTableException(e.getMessage() + " in " + stmt.toString(), e);
+        }
+
+    }
+
+    /**
+     * Check the table is exsists.
+     *
+     * @return boolean isTableExists.
+     */
+    public boolean isTableExist(PreparedStatement stmt) {
+        try {
+            stmt.executeQuery();
+            return true;
+        } catch (SQLException e) {
+            throw new RDBMSTableException("Table '" + "this.tableName" + "' assumed to not exist since its" +
+                    " existence check resulted " + "in exceptions " + e.getMessage() + " in " + stmt.toString(), e);
```

Comment:
No need to include e.getMessage() since stackTrace is passed.. Please check this in other places also.